Utils/Dataflow/071_esConsistency/consistency.py

```diff
@@ -160,6 +160,11 @@ def process(stage, message):
         if field in data:
             del data[field]
 
+    # Fields starting with an underscore are service fields. Some of them are
+    # treated in special way (see _id above). Service fields should not be
+    # checked, so they are removed.
+    data = {field: data[field] for field in data if field[0] != '_'}
```

Comment:
Maybe:
```
data = {field: ... if not field.startswith('_')}
```

Not that it will make anything better (it will only make a difference in case of empty string in `field` -- which is just as possible as non-string `field`, that will break both versions), just may look a bit more Pythonic.

---

I remember us discussing the usage of `startswith()` vs. checking leading substring, yet didn\`t decide anything specific. In particular, we even found out that `startswith()` may work slowlier that simple comparisson in some cases (and I believe this might be one of them), so maybe it doesn\`t worth changing.

Reply:
Checked the discussion - it seems `startswith()` is slower for short strings, which, I believe, includes our case (examples where  `startswith()` is faster operate with strings of 20000 symbols each). Therefore, this should be left as-is.